alphastats/gui/utils/analysis.py

```diff
@@ -0,0 +1,155 @@
+"""Module providing frontend widgets for gathering parameters and mapping them to the actual analysis."""
+
+from abc import ABC, abstractmethod
+from collections import defaultdict
+
+import streamlit as st
+
+from alphastats.DataSet import DataSet
+from alphastats.keys import Cols
+from alphastats.plots.VolcanoPlot import VolcanoPlot
+
+
+class Analysis(ABC):
+    """Abstract class for analysis widgets."""
+
+    def __init__(self, dataset):
+        self._dataset: DataSet = dataset
+        self._parameters = defaultdict(lambda: None)
+
+    @abstractmethod
+    def show_widget(self):
+        """Show the widget and gather parameters."""
+        pass
+
+    @abstractmethod
+    def do_analysis(self):
+        """Perform the analysis.
+
+        Returns a tuple(figure, analysis_object, parameters) where figure is the plot,
+        analysis_object is the underlying object, parameters is a dictionary of the parameters used.
+        """
+        pass
+
+
+class GroupCompareAnalysis(Analysis, ABC):
+    """Abstract class for group comparison analysis widgets."""
+
+    def show_widget(self):
+        """Gather parameters to compare two group."""
+
+        metadata = self._dataset.metadata
+
+        default_option = "<select>"
+        custom_group_option = "Custom group from samples .."
+
+        grouping_variable = st.selectbox(
+            "Grouping variable",
+            options=[default_option]
+            + metadata.columns.to_list()
+            + [custom_group_option],
+        )
+
+        column = None
+        if grouping_variable == default_option:
+            st.stop()  # TODO: using stop here is not really great
```

Comment:
Why? Because we might some logic down the line?

Reply:
I think we should not stop here. Otherwise if might break other part of the page, that would only be shown afterwards.

Reply:
this was resolved when adressing your feedback on the webapp (to always show all input fields)